lib/IDE/SyntaxModel.cpp

```diff
@@ -479,8 +476,19 @@ std::pair<bool, Expr *> ModelASTWalker::walkToExprPre(Expr *E) {
 
       pushStructureNode(SN, E);
     }
+  };
+
+  if (auto *ParentTupleExpr = dyn_cast_or_null<TupleExpr>(Parent.getAsExpr())) {
+    addCallArgExpr(E, ParentTupleExpr);
+  } else if (auto *ParentOptionalExpr = dyn_cast_or_null<OptionalEvaluationExpr>(Parent.getAsExpr())) {
+    if (auto *ParentTupleExpr = dyn_cast_or_null<TupleExpr>(ParentOptionalExpr->getSubExpr())) {
```

Comment:
We can push the if statement to the closure to reduce duplication, right?

Reply:
Also, could we have some comments here describing these two cases?

Reply:
> We can push the if statement to the closure to reduce duplication, right?

I've tried that, but I actually think it makes it less legible, because we lose that `if/else if` that is important to understand the two cases IMO.

Reply:
> Also, could we have some comments here describing these two cases?

I've added some comments in 96d6261. Let me know what you think!